scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts

```diff
@@ -56,6 +60,32 @@ export function AuthoringTopbarDirective(TranslationService, privileges, authori
             scope.isTranslationAvailable = function() {
                 return TranslationService.checkAvailability(scope.item);
             };
+
+            scope.extraActionsFromExtensions = [];
+
+            scope.triggerActionFromExtension = (actionToTrigger: IArticleAction) => {
+                actionToTrigger.onTrigger();
+            };
+
+            Object.keys(extensions).forEach((extension) => {
+                const {activationResult} = extensions[extension];
+
+                if (
+                    activationResult &&
+                    activationResult.contributions &&
+                    activationResult.contributions.entities &&
+                    activationResult.contributions.entities.article &&
+                    activationResult.contributions.entities.article.getExtraActions
+                ) {
+                    activationResult.contributions.entities.article.getExtraActions(scope.item)
+                        .then((actions) => {
+                            scope.extraActionsFromExtensions = [
```

Comment:
Use `flatMap` to get actions from all extensions at once as it's done in other places. Otherwise, you'll be triggering multiple re-renders when getting actions from each extension.